pkg/security/telemetry/telemetry.go

```diff
@@ -55,8 +49,33 @@ func (c *ContainersTelemetry) ReportContainers(metricName string) {
 			}
 		}
 
-		c.Sender.Gauge(metricName, 1.0, "", []string{"container_id:" + container.ID})
+		c.TelemetrySender.Gauge(metricName, 1.0, []string{"container_id:" + container.ID})
 	}
+	c.TelemetrySender.Commit()
+}
+
+// SimpleTelemetrySender is an abstraction over what is needed for the container telemetry
+// the main goal is to be able to use it with a dogstatsd client or a SenderManager's default sender
+type SimpleTelemetrySender interface {
+	Gauge(name string, value float64, tags []string)
+	Commit()
+}
+
+type statsdSTS struct {
+	sci statsd.ClientInterface
+}
+
+func (s *statsdSTS) Gauge(name string, value float64, tags []string) {
+	_ = s.sci.Gauge(name, value, tags, 1.0)
+}
 
-	c.Sender.Commit()
+func (s *statsdSTS) Commit() {
+	// nothing to do here
+}
```

Comment:
I assume that we keep `Commit` for function to preserve the interface. Do we plan of adding functionality in the future?

Reply:
We keep commit to keep the semantic of the sender manager, the statsd version doesn't need it. I don't see any reason this would change